### PR TITLE
Add Rod_Cutting DP problem, Ternary Search and Update documentation.md

### LINKED ISSUE #36: Implement Ternary Search
- Put your file in the right folder (/Algorithms/Search).
- Do corresponding additions in the documentation.
- Comment your code where necessary.

### LINKED ISSUE #39: Segment Trees is not accessible through Data Structures Implementation
Segment Trees code page is not accessible when one goes through Data Structures implementation and click on Segment Trees. It is redirecting to Fenwick Trees.

Screenshot (1191)

### LINKED ISSUE #40: Implement Rod Cutting DP Problem

### PR DESCRIPTION
**Pull request by Mirnal1409**
# Description
Added the Rod Cutting - DP Question
Added Ternary Search
Fixed #40 
Fixed #39 
fixed #36 

Updated documentation.md

## Type of change

Please check options that are relevant to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New Data Structure 
- [ *] New Algorithm

# Checklist:

- [ *] I have squashed my commits
- [ *] I have commented my code, particularly in hard-to-understand areas
- [ *] I have made corresponding changes to the documentation
- [ *] My code is a working bug-free code

### CODE REVIEW

File: Algorithms/Dynamic Programming/rod_cutting(DP).cpp [hunk 1]
```diff
@@ -0,0 +1,54 @@
+
```
Comment (Manvityagi) [inline Algorithms/Dynamic Programming/rod_cutting(DP).cpp:1]: Please provide the link to the question in the comment form.

File: Algorithms/Dynamic Programming/rod_cutting(DP).cpp [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    return t[n][Max_len];
+}
+
+/* Driver program to
+test above functions */
+
+int main()
+ {
+
```
Comment (Manvityagi) [inline Algorithms/Dynamic Programming/rod_cutting(DP).cpp:34]: indent your code properly, remove extra blank lines.

File: Algorithms/Search/ternarySearch.cpp [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  sort(arr,arr+n);  // we sort the array provided
+
+  int res=ternarySearch(0,n-1,elem,arr);
+  if(res==-1)
+  cout<<"Element not found";
+  else
+  cout<<"Index of"<<elem<<"is :"<<res;
+
+
+}
```
Comment (Manvityagi) [inline Algorithms/Search/ternarySearch.cpp:57]: You may send different PR for this algorithm.

**Review by Manvityagi — CHANGES_REQUESTED** [review]
Everything else seems perfect, just a few changes that I have commented. Please do them to have your PR merged.

**Review by Manvityagi — CHANGES_REQUESTED** [review]
Also, please open an issue for adding rod cutting. 
And while sending the PR, mention that particular issue, (for example fixed #2) where 2 - Issue number being solved by your PR.

File: Algorithms/Search/ternarySearch.cpp (hunk 3 above)
Comment (Mirnal1409) [reply Algorithms/Search/ternarySearch.cpp:57]: ok I will do this.

File: Algorithms/Dynamic Programming/rod_cutting(DP).cpp (hunk 2 above)
Comment (Mirnal1409) [reply Algorithms/Dynamic Programming/rod_cutting(DP).cpp]: Done ,please review the changes in new commit

File: Algorithms/Dynamic Programming/rod_cutting(DP).cpp (hunk 1 above)
Comment (Mirnal1409) [reply Algorithms/Dynamic Programming/rod_cutting(DP).cpp]: Done in new commit please review the changes

**Review by Manvityagi — APPROVED** [review]
Looks good @Mirnal1409
